main/src/addins/MonoDevelop.Debugger.VSCodeDebugProtocol/MonoDevelop.Debugger.VsCodeDebugProtocol/VSCodeDebuggerSession.cs

```
@@ -280,8 +278,8 @@ protected void HandleEvent (object sender, EventReceivedEventArgs obj)
 					}
 					currentThreadId = body.ThreadId ?? -1;
 					//TODO: what happens if thread is not specified?
-					args.Process = OnGetProcesses () [0];
-					args.Thread = GetThread (args.Process, (long)body.ThreadId);
+					args.Process = GetProcesses () [0];
+					args.Thread = args.Process.GetThreads ().Single (t => t.Id == currentThreadId);
```

It would fail on next line anyway, feels better if it fails fast